src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -957,8 +960,8 @@ public Location getCompassTarget()
 	@Override
 	public InetSocketAddress getAddress()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		// Assuming the "client" and the "server" are on the same machine.
+		return new InetSocketAddress(InetAddress.getLoopbackAddress(), 25565);
```

Comment:
I highly doubt this is right, but isn't the port here not the client's source port instead of the destination port? If both client and server are on the same machine, and the server's port is 25565, it should be impossible for the client to the server using source port 25565.
The source port should be some randomized number probably within the range of 32768 - 65535, and should unique for all connections using that IP address.

In fact, we could probably also randomize the IP address. Perhaps best to use an IP within a range that is [reserved for documentation and examples](https://en.wikipedia.org/wiki/Reserved_IP_addresses):
 - 192.0.2.0/24
 - 198.51.100.0/24
 - 203.0.113.0/24

Reply:
Yup, you're correct. Implemented.